internal/controllers/password_reset_controller.go

```diff
@@ -0,0 +1,71 @@
+package controllers
+
+import (
+	"github.com/gin-gonic/gin"
+	"github.com/wwi21seb-projekt/server-beta/internal/customerrors"
+	"github.com/wwi21seb-projekt/server-beta/internal/models"
+	"github.com/wwi21seb-projekt/server-beta/internal/services"
+	"net/http"
+)
+
+type PasswordResetControllerInterface interface {
+	PasswordReset(c *gin.Context)
+	SetNewPassword(c *gin.Context)
+}
+
+type PasswordResetController struct {
+	passwordResetService services.PasswordResetServiceInterface
+}
+
+// NewPasswordResetController can be used as a constructor to return a new PasswordResetController "object"
+func NewPasswordResetController(passwordResetService services.PasswordResetServiceInterface) *PasswordResetController {
+	return &PasswordResetController{passwordResetService: passwordResetService}
+}
+
+// PasswordReset triggers a password reset process for the user
+func (controller *PasswordResetController) PasswordReset(c *gin.Context) {
+	// Read body
+	var passwordResetRequestDTO models.PasswordResetRequestDTO
+
+	if c.ShouldBindJSON(&passwordResetRequestDTO) != nil {
+		c.JSON(http.StatusBadRequest, gin.H{
+			"error": customerrors.BadRequest,
+		})
+		return
+	}
```

Comment:
Der Request wurde in Postman anders definiert. Der Username wird aus der URL ausgelesen und das DTO fehlt nicht